src/middlewared/middlewared/plugins/datastore/write.py

```diff
@@ -29,7 +28,11 @@ class Config:
                     insert.setdefault(column.name, '')
 
         await self.middleware.call('datastore.execute_write', table.insert().values(**insert))
-        pk = (await self.middleware.call('datastore.fetchall', 'SELECT last_insert_rowid()'))[0][0]
+        pk_column = self._get_pk(table)
+        if isinstance(pk_column.type, (types.String, types.TEXT)):
```

Comment:
The only type for primary key being rowid is integer, for all other types (including e.g. bigint sqlite requires us to specify pk explicitly)